gips/0081-indexing-payments.md

```diff
@@ -0,0 +1,165 @@
+
+### Arbitration and disputes
+
+Since payment collections are tied to "amount of work done" and this is reported by the indexers themselves, there needs to be a mechanism to punish malicious reporting. We propose to use slashing. For this to be fair we need the "amount of work done" to be verifiable by an arbitrator in the case of a dispute.
+
+### Amount of work done
+
+As an initial iteration of the "amount of work done" proxy we propose to use indexed blocks and subgraph entities. In particular the amount of payment to be collected by indexers shall be equal to:
+
+- A price per block indexed (with a different price for each chain). These are charged incrementally, i.e. every time a POI is posted, the indexer collects a payment for the new blocks indexed since the last POI.
+- A price per extracted entity of the subgraph (same price for each chain). These are charged on a recurring basis per unit of time, for the total number of entities on the subgraph at the time the POI is posted multiplied by the time since the last POI.
+
+We understand that this approach is vulnerable to some forms of abuse. For example, malicious users can deploy subgraphs that are expensive to index but amount to a comparatively small "amount of work done" proxy fee. We suggest that indexers stop serving such subgraphs as soon as possible, to minimize the damage done.
+
+We anticipate needing to perform research into alternative approaches in the near future. We theorize that "amount of work done" may be modeled as a **subgraph gas** unit down the line, similar to ethereum gas.
+
+### Minimum Viable Product
+
+While this proposal defines a smart contracts based solution using the Subgraph Service on Graph Horizon, we also propose an MVP that can be built without any smart contracts changes using off-chain interfaces between Indexers and Gateways. This will allow us to implement parts of the solution that will be useful for the full proposal implementation, will also allow us to learn from seeing the mechanism in operation, and will help define the user interfaces and pricing approaches for subgraph developers that can be built on top of the protocol. As the MVP does not require changes to the on-chain protocol, it technically does not require a formal GIP approval, but we present it here alongside the full proposal to gather consensus towards this approach and to invite Indexers to participate in this initial stage and provide feedback.
+
+In this MVP, we will roll out a new gateway component (the “Dipper”) that will be controlled with a CLI. Gateway operators can be onboarded to this CLI and use it to add subgraphs to the list of supported subgraphs. Once a subgraph is in the list, the Dipper will take care of finding Indexers and setting up indexing agreements with them.
```

Comment:
Looking at this again, where does the name Dipper come from? Was it coined in terms of "DIPs", which we are moving away from? Is there a better option for people to intuitively understand what it is?